app/src/main/java/com/sdpteam/connectout/post/model/PostDataSource.java

```diff
@@ -0,0 +1,63 @@
+package com.sdpteam.connectout.post.model;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import com.sdpteam.connectout.utils.Result;
+
+public interface PostDataSource {
+
+    /**
```

Comment:
I would suggest adding the method `getUniqueId() ` since a Post has knowledge of its ID, it will be necessary to create it in the view.
For reference please check out `EventCreatorActivity` line 78 and its Model and ViewModel.

Reply:
In fact I don't want to do like in EventCreatorActivity because it is not good.
In OOP we should follow the "tell don't ask" principle (check out for instance https://softwareengineering.stackexchange.com/a/157527 or https://medium.com/vattenfall-tech/tell-dont-ask-learn-to-talk-to-your-objects-45d7c4aa61fe)

It is an abstraction leak to expose this way to return a new unique id that will be used as the id before when creating a new item `dataSource.savePost(new Post(dataSource.getUniqueNewId(), ...))`.
It is the savePost that will handle this managing of unique ids.

```
    /**
     * Create or update a post identified with Post.id
     *
     * @return a completable future of a Result (which says if it is successful or not)
     */
    CompletableFuture<Result<Void>> savePost(Post post);
```
By writing "Create or update a post identified with Post.id" it wants to say the `savePost` will simply check if the post.id value is null then create a new id and save it there, or if the post.id  value is not null check if it exists and update the entry. But yes I will specify this better in the JavaDoc.

Reply:
I  see your point and agree. Thank you for reminding me of the principle !
We should add an issue to fix Event this way.